src/calculateRentalCost.js

```diff
@@ -3,8 +3,25 @@
  *
  * @return {number}
  */
+
+const COST_PER_DAY = 40;
+const MEDIUM_PERIOD = 3;
+const MEDIUM_PERIOD_DISCOUNT = 20;
+const LONG_PERIOD = 7;
+const LONG_PERIOD_DISCOUNT = 50;
+
 function calculateRentalCost(days) {
-  // write code here
+  const rentalCost = days * COST_PER_DAY;
+
+  if (days >= LONG_PERIOD) {
+    return rentalCost - LONG_PERIOD_DISCOUNT;
+  }
+
+  if (days >= MEDIUM_PERIOD) {
+    return rentalCost - MEDIUM_PERIOD_DISCOUNT;
```

Comment:
The logic here applies a medium period discount even if the days qualify for a long period discount. Consider restructuring the conditions to ensure that only the highest applicable discount is applied. You might want to check for the long period first, then the medium period, to ensure the correct discount is applied.